wgpu-types/src/lib.rs

```diff
@@ -3540,6 +3540,47 @@ impl TextureFormat {
     pub fn is_srgb(&self) -> bool {
         *self != self.remove_srgb_suffix()
     }
+
+    /// Returns the theoretical memory footprint of a texture with the given format and dimensions.
+    ///
+    /// Actual memory usage may greatly exceed this value due to alignment and padding.
+    #[must_use]
+    pub fn theoretical_memory_footprint(&self, size: Extent3d) -> u64 {
+        let (block_width, block_height) = self.block_dimensions();
+
+        let block_size = self.block_copy_size(None);
+
+        let approximate_block_size = match block_size {
+            Some(size) => size,
+            None => match self {
+                // One f16 per pixel
+                Self::Depth16Unorm => 2,
+                // One u24 per pixel, padded to 4 bytes
+                Self::Depth24Plus => 4,
+                // One u24 per pixel, plus one u8 per pixel
+                Self::Depth24PlusStencil8 => 4,
+                // One f32 per pixel
+                Self::Depth32Float => 4,
+                // One f32 per pixel, plus one u8 per pixel, with 3 bytes intermediary padding
+                Self::Depth32FloatStencil8 => 8,
```

Comment:
thought: Sorta weird that this is padded to an alignment of 8, but others aren't. 🤔 Wish I knew more about this domain!

Reply:
This is some random bit of knowledge about how... nvidia packs these buffers, iirc. It is two elements, one f32, one u8, but with align 4 on the structure, so you get 3 bytes of padding.